src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -671,6 +671,9 @@ impl Step for Rustdoc {
 
         // don't create a stage0-sysroot/bin directory.
         if target_compiler.stage > 0 {
+            // Due to LTO a lot of debug info from C++ dependencies such as jemalloc can make it into
+            // our final binaries
+            compile::strip_debug(builder, target, &tool_rustdoc);
```

Comment:
This should also be done conditionally only if no compiler debuginfo is requested, as someone might want to debug rustdoc.